### PR TITLE
Completion of Week 1 + 2 Walkthroughs and a preliminary website design

### PR DESCRIPTION
**Pull request by mmurray22**
This PR includes both the beginning changes the walkthroughs requested, as well as some serious developmental changes I added today. At this point, this branch embodies a basic, working website with minimal information and basic functionality.

### CODE REVIEW

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: uber nit - Although I can't find specific guidance in the [html style guide](https://google.github.io/styleguide/htmlcssguide.html) - all of their examples don't have whitespace between the start-tag and the content. I would drop the space.

File: portfolio/src/main/webapp/style.css
Comment (ccondit) [inline portfolio/src/main/webapp/style.css]: nit - Per the [style guide](https://google.github.io/styleguide/htmlcssguide.html#Declaration_Order):
`Put declarations in alphabetical order in order to achieve consistent code in a way that is easy to remember and maintain.`

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: I think `br` is an empty element tag. Would this be better modeled as a list?

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: nit - let's avoid committing commented out code. We can always recover it from the repository history.

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: You're a vi user, i see 😃

File: portfolio/src/main/webapp/script.js
Comment (ccondit) [inline portfolio/src/main/webapp/script.js]: uber nit - drop the whitespace at the beginning of this string

File: portfolio/src/main/webapp/style.css
Comment (ccondit) [inline portfolio/src/main/webapp/style.css]: Can you [omit the leading zeros](https://google.github.io/styleguide/htmlcssguide.html#Leading_0s) in the rgba values?

**Review by ccondit — COMMENTED** [review]
nit - s/basuc/basic/ in the commit message?

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Thanks for the input! I looked through my current HTML file and made sure all spaces were dropped.

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Oops, lol my bad. Fixed!

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Noted (and updated)! Thanks for the feedback!

File: portfolio/src/main/webapp/script.js
Comment (mmurray22) [reply portfolio/src/main/webapp/script.js]: Noted (and updated)! Thanks for the feedback!

File: portfolio/src/main/webapp/style.css
Comment (mmurray22) [reply portfolio/src/main/webapp/style.css]: Noted (and updated)! Thanks for the feedback!

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Yes it would! Updated!

File: portfolio/src/main/webapp/style.css
Comment (mmurray22) [reply portfolio/src/main/webapp/style.css]: Found an alternative! Thanks for pointing that out!

Comment (ccondit) [reply portfolio/src/main/webapp/style.css]: Did the commit message update? Looks like it still has the typo. Maybe try [this](https://help.github.com/en/github/committing-changes-to-your-project/changing-a-commit-message).

image

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [reply portfolio/src/main/webapp/index.html]: Cool! Mind fixing the indentation for the list?

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: nit - this div is missing its end tag.

File: portfolio/src/main/webapp/script.js
Comment (ccondit) [inline portfolio/src/main/webapp/script.js]: Per style guide: [Do not add semicolons after methods](https://google.github.io/styleguide/jsguide.html#formatting-class-literals).

Comment (ccondit) [inline portfolio/src/main/webapp/script.js]: Per [style guide](https://google.github.io/styleguide/jsguide.html#formatting-column-limit) there should be an 80 character column limit. Mind reformatting here and below?

**Comment by ccondit** [comment]
Also, generally this PR has expanded beyond the initial description - mind updating it to reflect the scope of the current commits?

**Review by ccondit — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js
Comment (mmurray22) [reply portfolio/src/main/webapp/script.js]: Noted (and updated). Thanks! Does this only apply to the javascript files?

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Noted (and updated). Thanks!

File: portfolio/src/main/webapp/script.js
Comment (mmurray22) [reply portfolio/src/main/webapp/script.js]: Noted (and updated)! Thanks!

File: portfolio/src/main/webapp/style.css
Comment (mmurray22) [reply portfolio/src/main/webapp/style.css]: I'm not sure if I can change the previous commit's wording unfortunately

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Just fixed indentation in the index.html!

File: portfolio/src/main/webapp/index.html
Comment (kaveh096) [inline portfolio/src/main/webapp/index.html]: The indentations look off here. But it looks ok on file view. it's only off in commit view. i think it's because you're mixing tab and space for indentation.

Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: yeah hm  that's weird...I don't see that in my original code lol

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Hello Chris, would we be able to resolve this comment? I think I've addressed this already, and it keeps showing up as "not addressed". Thanks in advance!

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [reply portfolio/src/main/webapp/index.html]: This is how it's rendering for me: image

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: Did you mean for this text to fall outside of the `<span>` (and with all the spans below)?

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Ah I see..the < / p > tag is out of place

Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: I will commit my index.html file to this branch, although there have been some changes to it-I think once I sit down & make sure this vim setup doesn't use tabs as default spacing the spacing will get better. Thanks for the info!

Comment (kaveh096) [reply portfolio/src/main/webapp/index.html]: This is how it looks like on the commit view for me: 
image

Anyways, let's clos this thread. Cause it looks fine on file view.

File: portfolio/src/main/webapp/script.js
Comment (kaveh096) [reply portfolio/src/main/webapp/script.js]: It's optional: https://google.github.io/styleguide/htmlcssguide.html#HTML_Line-Wrapping

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [reply portfolio/src/main/webapp/index.html]: Looks like the diff has shifted since this commit. I'll follow up on the current snapshot...

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: There are multiple divs now with id="button-div". I think element ids should be unique across this document.

File: portfolio/src/main/webapp/script.js
Comment (ccondit) [inline portfolio/src/main/webapp/script.js:62]: One thing I was thinking about is that it's a bit cumbersome to store this content in string variables in javascript when it really makes more sense to keep it in the HTML.

Let's no block on this, but consider following up to migrate this into the DOM and have these javascript functions just reveal the content that was initially hidden...

**Review by ccondit — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: I believe my aim in adding that uniform id was to apply style changes to all those buttons uniformly. However, since I don't have any other buttons to style, I can definitely just use the button tag as a classifier instead. Thanks!

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Yes for some odd reason that is how the Material Design sample code looked...I can definitely put the names inside the tags for consistency though! Thank you!

File: portfolio/src/main/webapp/script.js
Comment (mmurray22) [reply portfolio/src/main/webapp/script.js:62]: Thank you for the suggestion, and I will make sure to incorporate that in my next PR

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: Is the `background()` function defined?

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: nit - same comment about whitespace between the tags and the content (and below).

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: looks like some git command worked the way in 😀

**Review by ccondit — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Yes it is! In the other branch

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: oof my bad! thanks!

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Noted! Checking for this right now! thanks!

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [reply portfolio/src/main/webapp/index.html]: Let's omit this in the current commit until that branch is ready. With this pull request when this page loads, `background` is undefined: 
image

The goal here is to keep the main branch always in a consistent state.

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: In a similar vein to the above comment, the `blog` method doesn't exist, either. I would drop this `onclick` in this PR and then add it in a follow up.

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: nit - now that this isn't a `p` element anymore, the `para` is a mismatch. Can we choose a more specific id that communicates what this div will contain?

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: This indentation still looks off in the review tool: 
image
In particular the list elements are indented less than the parent `ul`.

**Review by ccondit — CHANGES_REQUESTED** [review]
Cool! Almost there, I think :) Just a few more changes...

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: sounds good! Thanks!

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: sounds good! thanks!

File: portfolio/src/main/webapp/index.html
Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: do you have any recommendations for fixing this? I can't see it on my end :(

Comment (mmurray22) [reply portfolio/src/main/webapp/index.html]: Okay I edited it directly in github...hopefully that works lol

File: portfolio/src/main/webapp/script.js
Comment (ccondit) [inline portfolio/src/main/webapp/script.js:35]: nit - there's an extra space between `my` and `-info`.

File: portfolio/src/main/webapp/index.html
Comment (ccondit) [reply portfolio/src/main/webapp/index.html]: Interesting - the indentation still looks off in the diffing tool:
image

But let's not block on this for this review. I've asked the intern team for input on what normalization tools they'd like us to use. We can follow up offline later to resolve this for the team.

**Review by ccondit — APPROVED** [review]
This LGTM. We'll follow up later on general formatting tools with the rest of the team.

Please also give Kaveh a chance to review before you merge.

**Pull request merged**
